server/etcdmain/config.go

```diff
@@ -165,6 +170,20 @@ func (cfg *config) parse(arguments []string) error {
 		return perr
 	}
 
+	var warningsForDeprecatedFlags []string
+	cfg.cf.flagSet.Visit(func(f *flag.Flag) {
+		if msg, ok := deprecatedFlags[f.Name]; ok {
+			warningsForDeprecatedFlags = append(warningsForDeprecatedFlags, msg)
+		}
+	})
+	if len(warningsForDeprecatedFlags) > 0 {
+		if lg := cfg.ec.GetLogger(); lg != nil {
+			for _, msg := range warningsForDeprecatedFlags {
+				lg.Warn(msg)
+			}
+		}
+	}
```

Comment:
It seems that it only checks the command line flags. We should check the config file input as well. Proposed actions:
- Move this code snip into `configFromCmdLine`;
- add similar check in `configFromFile`